editors/code/src/inlay_hints.ts

```diff
@@ -5,19 +5,27 @@ import { Ctx, sendRequestWithRetry } from './ctx';
 
 export function activateInlayHints(ctx: Ctx) {
     const hintsUpdater = new HintsUpdater(ctx);
-    vscode.window.onDidChangeVisibleTextEditors(async _ => {
-        await hintsUpdater.refresh();
-    }, ctx.subscriptions);
-
-    vscode.workspace.onDidChangeTextDocument(async e => {
-        if (e.contentChanges.length === 0) return;
-        if (e.document.languageId !== 'rust') return;
-        await hintsUpdater.refresh();
-    }, ctx.subscriptions);
-
-    vscode.workspace.onDidChangeConfiguration(_ => {
-        hintsUpdater.setEnabled(ctx.config.displayInlayHints);
-    }, ctx.subscriptions);
+    vscode.window.onDidChangeVisibleTextEditors(
+        async _ => hintsUpdater.refresh(),
```

Comment:
I'm probably missing something, but isn't this floating? I would have expected either `_ => hintsUpdater.refresh()` or `_ => { await hintsUpdated.refresh(); }` Compare with the two below.

Reply:
No this is not floating, we return this promise. When we are returning a Promise from async func, it will be automatically flattened from Promise Promise<...>> to just Promise <...>>. Though yes, we can remove async keyword, but I left it there just to emphasize that the operation here is asynchronous.

Just an example: 
```ts
await Promise.resolve(Promise.resolve(Promise.resolve(42))) === 42
```
You can check it in your browser console

Reply:
> When we are returning a Promise from async func, it will be automatically flattened from `Promise<Promise<...>>` to just `Promise <...>`.

That's just... :confused:.

Reply:
Yeah, there is even a dedicated tslint rule [`no-return-await`](https://palantir.github.io/tslint/rules/no-return-await), but I'd rather not turn it on in this project, because it appears to be controversial on my experience...